python/pyspark/ml/wrapper.py

```diff
@@ -82,13 +71,16 @@ def _transfer_params_to_java(self):
                 pair = self._make_java_param_pair(param, paramMap[param])
                 self._java_obj.set(pair)
 
-    def _transfer_params_from_java(self):
+    def _transfer_params_from_java(self, withParent=False):
         """
         Transforms the embedded params from the companion Java object.
         """
         sc = SparkContext._active_spark_context
+        parent = self._java_obj.uid()
         for param in self.params:
             if self._java_obj.hasParam(param.name):
+                if withParent:
+                    param.parent = parent
```

Comment:
We should also copy ```parent``` from peer java object if the python object is instantiated by ```load```.